spec/mailers/notify_mailer_spec.rb

```diff
@@ -16,7 +16,7 @@
 
     it "renders proper sender" do
       expect(email.from).to eq([SiteConfig.email_addresses[:default]])
-      expected_from = "#{SiteConfig.community_name} Community <#{SiteConfig.email_addresses[:default]}>"
+      expected_from = "#{SiteConfig.community_name} <#{SiteConfig.email_addresses[:default]}>"
       expect(email["from"].value).to eq(expected_from)
     end
```

Comment:
This (and all "community"-related `notify_mailer_spec` tests) is failing because `email["from"].value` returns `DEV (local) Community`, but I changed the value of `expected_from` to _not_ include the appended `Community` and I cannot figure out why `.value` still returns it _with_ the appended `Community` in the test. After looking into the mailer-related code and throwing `binding.pry`s in, I was unable to trace this `.value` back to anything that stood out as the culprit. Am I overthinking this? Should `Community` still remain appended in the spec? 😅 🤔

Reply:
You're not overthinking this :) I believe your changes aren't showing up here the way we'd expect because there is a spot in the spec configuration/setup where we _explicitly_ override the "from" of an email 😅 

Check out `application_mailer.rb`, specifically these lines:
https://github.com/forem/forem/blob/23235c2276242a0ccd8213bcf3154d09507b541b/app/mailers/application_mailer.rb#L12-L15

Rails [lets you override](https://api.rubyonrails.org/classes/ActionMailer/Base.html) the default hash in the `ApplicationMailer`, which is what we're doing here (not sure _why_, but I am 99% sure it's just a remnant of when we used to hardcode `"Community"` everywhere 😛).

The real culprit here is this bit though, which is invoked in line `13` of the `ApplicationMailer` I linked to above:
https://github.com/forem/forem/blob/23235c2276242a0ccd8213bcf3154d09507b541b/app/mailers/application_mailer.rb#L17-L19

We pass in a `topic`, and then render `topic` between `community_name` and the default `email_address`. And....it looks like in line 13 of the `ApplicationMailer`, we were passing in the string `"Community"` as the topic 😩 

That's why it kept on showing up in those tests, even though you had actually removed it from the code everywhere else.

IMO, you could remove that `topic` parameter from the method signature entirely, so that the `email_from` method looks like this:
```ruby
def email_from
   "#{SiteConfig.community_name} <#{SiteConfig.email_addresses[:default]}>"
end
```

The only other thing I'd add here is that we _may_ want to change `community_name` on the DEV Forem instance to be `DEV Community` (? Maybe check on that with community support/@benhalpern) if we want the "from" in _our DEV emails_ to continue to be `"DEV Community <yo@dev.to>"` on production, rather than `"DEV <yo@dev.to>`, which is what it would change to as a result of my above suggestions :)

Reply:
Thank you so much for the sanity check! `#{topic}` is really hidden in there. 🔎  

> The only other thing I'd add here is that we may want to change community_name on the DEV Forem instance to be DEV Community (? Maybe check on that with community support/@benhalpern) if we want the "from" in our DEV emails to continue to be "DEV Community <yo@dev.to>" on production, rather than "DEV <yo@dev.to>,

YES! This is something I was thinking about as well--great suggestion! 😄

Reply:
_Update_: I ended up reverting the changes I made to `application_mailer.rb` and the `notify_mailer_spec.rb` after a bit more digging. It actually appears that the `topic` parameter _is_ necessary and is used in another file and spec, `digest_mailer.rb and `digest_mailer_spec.rb`, for reasons other than appending "Community" to the `community_name`. Below is a snippet from the spec, which shows that the `topic` parameter is used, but not to append "Community" like in the `notify_mailer`:
```
"#{SiteConfig.community_name} Digest <#{SiteConfig.email_addresses[:default]}>"
```
That being said, it looks like the default of "Community" is overridable, so I don't think it'll be harmful keeping it as-is in the `application_mailer.rb` or `notify_mailer_spec.rb`, but I'm more than open to suggestions on how to better generalize this if possible! cc @vaidehijoshi @benhalpern

Reply:
Interesting, and good digging! I see two possible paths here:

1. We allow a optional parameter to be passed into email_from, and remove "Community", which would allow us to keep the concept of "from [some forem community name] email".
2. We remove the concept of "Digest"—I am not sure what the digest feature looks like for other Forems (maybe @benhalpern has more historical context here). Seems like we could just have all the "from" emails come from one singular "account" for simplicity's sake (it might also just be easier for Forem admins to keep mentally keep track of).

I would be fine either way, and personally think option 2 makes more sense holistically, but I also think that is out of the scope for this PR.

In the meantime, I do think we should remove "Community", but allow an optional param to be passed in (I think that's what you had previously, @juliannatetreault!). The reason I'd advocate to remove "Community" as the default hard-coded value is that if someone names their community "The Llama Community", then we'd have their emails coming from "The Llama Community Community". 😆

Reply:
This discussion is gold! Thanks @juliannatetreault and @vaidehijoshi ❤️

Reply:
Thank _you_, @rhymes! 

**Update for those following this thread**: I was able to get the mailers and mailer-related specs to work by passing a default (`topic = ""`) to the `topic` param in `application_mailer.rb` and refactoring `#email_from` to be a conditional based upon the `topic`, if any, that is passed to the method. By doing this I was able to remove the default "Community" that was appended to the emails in `notify_mailer.rb` and was able to keep the appended "Digest" in `digest_mailer.rb`. Let me know what you all think about the approach! 😄 

cc @vaidehijoshi @benhalpern